src/main/java/com/exasol/bucketfs/Bucket.java

```diff
@@ -142,35 +176,69 @@ private String extractFirstPathComponent(final String path) {
      * Uploads a file from a given local path to a URI pointing to a BucketFS bucket. If the bucket URI ends in a slash,
      * that URI is interpreted as a directory inside the bucket and the original filename is appended.
      * </p>
+     * <p>
+     * This call blocks until the uploaded file is synchronized in BucketFs or a timeout occurs.
+     * </p>
      *
      * @param pathInBucket path inside the bucket
      * @param localPath    path of the file to be uploaded
+     * @throws TimeoutException      if the synchronization check takes too long
      * @throws InterruptedException  if the upload is interrupted
      * @throws BucketAccessException if the file cannot be uploaded to the given URI
      */
     // [impl->dsn~uploading-to-bucket~1]
     public void uploadFile(final Path localPath, final String pathInBucket)
+            throws InterruptedException, BucketAccessException, TimeoutException {
+        uploadFile(localPath, pathInBucket, true);
+    }
+
+    /**
+     * Upload a file to the bucket and block the call until it is synchronized.
+     * <p>
+     * Uploads a file from a given local path to a URI pointing to a BucketFS bucket. If the bucket URI ends in a slash,
```

Comment:
I think this information should also be in the user guide

Reply:
Added section: Uploading Files Into a "Directory"